EIPS/eip-6927.md

```diff
@@ -0,0 +1,166 @@
+
+    /// @dev Retrieve a combo's ingredients.
+    function getIngredients(
+        uint256 comboId
+    ) external view returns (Ingredient[] memory);
+
+    /// @dev Approves `spender` to use `msg.sender`'s NFTs when minting a combo.
+    /// The approval allows the spender to use the specified NFTs within their allowance limits.
+    /// Only ERC-721 NFTs are approvable.
+    /// @param tokenAddresses The NFT's collection addresses
+    /// @param tokenIds The NFT identifiers
+    /// @param allowances The maximum number of uses
+    function approveCombo(
+        address spender,
+        address[] calldata tokenAddresses,
+        uint256[] calldata tokenIds,
+        uint256[] calldata allowances
+    ) external;
+}
+```
+
+## Rationale
+
+Firstly, the process of fusing existing NFTs to create new ones is highly diverse. The current NFT trading market uses a price discovery mechanism dominated by collection floor prices. To facilitate price discovery, it is essential to assign new NFTs created with different combination rules to separate collections. Therefore, we have extended the ERC-721 to support various combination rules.
+
+Secondly, the potential for customizable combination rules is immense, and maintaining simplicity and flexibility is a fundamental principle. Given the current NFT trading market’s price discovery mechanism, which is dominated by collection floor prices, we have positioned the collection as the basic unit of combination factors. Both ERC-721 and [ERC-1155](./eip-1155.md) are supported as combination factors. To avoid over-design, we have added only the most necessary parameters for each combination factor, such as upper and lower bounds and lock settings. Moreover, to prevent users from repeatedly creating combos using the same NFT without locking and destroying the scarcity of new assets, we have added usage limits for NFTs that do not require locking in the combination.
+
+Lastly, fusing NFTs results in an internal structure within combos. These ingredients are distinct from metadata that cannot be read by smart contracts. They can be read by other smart contracts and used in more native on-chain application scenarios, such as smart airdrops and raffles based on different ingredients of combos.
+
+## Backwards Compatibility
+
+This proposal is fully backwards compatible with the existing ERC-721 standard, extending the standard with new functions that do not affect the core functionality.
+
+## Test Cases
+
+Test cases should be developed to cover the following scenarios:
+
+* Minting a valid combo with ERC-721 and/or ERC-1155 NFTs.
+* Dismantling a combo and returning locked NFTs to their owners.
+* Setting combo rules and ensuring compliance during the minting - - process.
+* Approving ERC-721 NFTs for use in combos.
+* Querying combo rules and ingredients.
+
+## Reference Implementation 
+
+An implementation of this EIP will be provided upon acceptance of the proposal.
+
```

Comment:
```suggestion
<!--

## Test Cases

Test cases should be developed to cover the following scenarios:

* Minting a valid combo with ERC-721 and/or ERC-1155 NFTs.
* Dismantling a combo and returning locked NFTs to their owners.
* Setting combo rules and ensuring compliance during the minting - - process.
* Approving ERC-721 NFTs for use in combos.
* Querying combo rules and ingredients.

## Reference Implementation 

An implementation of this EIP will be provided upon acceptance of the proposal.

-->
```

These sections are optional, and if you use an HTML-style comment, the linter will make sure you come back to them before advancing statuses.